### DISCUSSION TITLE
How to know max table rows or display continous text (tail like) ?

### DISCUSSION THREAD

**Discussion by damsgithub** [Textualize/rich#1090] [Q&A] [answered]
Hello,

I use rich to display messages in a table (itself in a Layout), each message adding a row to the table. All is well until the displayed table is full and the added row no longer can appear in it, so it is not displayed at all. Is there a way to know when the table is "full", so that I can for example clear its content and start over ?

Or may be I should not use a table ? Is there a way to display continuous text messages in a particular Layout that will automatically scroll (tail like)?

thanks in advance

**Comment by willmcgugan** [comment, marked as answer]
You probably don't need a table here. If you create a renderable using the [console protocol](https://rich.readthedocs.io/en/latest/protocol.html) you can refer to `options.height` and return that many lines.

```python
class MyLog:
    def __rich_console__(self, console, options):
        for line in self.lines[-options.height:]:
            yield line
        
```

In the future I'll probably add a `rich.log.Log` class that has this functionality.

**Reply by damsgithub** [reply]
Thanks, I will try that.
